locations/spiders/beacon_and_bridge.py

```diff
@@ -1,57 +1,83 @@
 import scrapy
-
+import re
+from datetime import datetime
+from locations.hours import OpeningHours
 from locations.items import GeojsonPointItem
-
-URL = "http://www.beaconandbridge.com/wp-admin/admin-ajax.php"
-HEADERS = {
-    "Accept-Language": "en-US,en;q=0.9,ru;q=0.6",
-    "Origin": "http://www.beaconandbridge.com",
-    "Accept-Encoding": "gzip, deflate, br",
-    "Accept": "application/json, text/plain, */*",
-    "Referer": "http://www.beaconandbridge.com/locations/",
-    "Connection": "keep-alive",
-    "Content-Type": "application/x-www-form-urlencoded; charset=UTF-8",
-}
+from locations.hours import DAYS
 
 
 class BeaconAndBridgeSpider(scrapy.Spider):
     name = "beacon_and_bridge"
     item_attributes = {"brand": "Beacon Bridge Market"}
-    allowed_domains = ["www.beaconandbridge.com"]
-    custom_settings = {"ROBOTSTXT_OBEY": False}
-
-    def start_requests(self):
-        form_data = {
-            "formdata": "addressInput=",
-            "lat": "43.0142978",
-            "lng": "-83.68935469999997",
-            "radius": "10000",
-            "action": "csl_ajax_onload",
-        }
-
-        yield scrapy.http.FormRequest(
-            url=URL,
-            method="POST",
-            formdata=form_data,
-            headers=HEADERS,
-            callback=self.parse,
-        )
+    start_urls = ["https://beaconandbridge.com/locations/"]
 
     def parse(self, response):
-        results = response.json()
-        for data in results["response"]:
+        address_list = response.xpath(
+            '//div[@id="section_2"]//div[contains(@class, "yes_heads")]'
+        )
+        for i, item in enumerate(address_list):
+            hour = (
+                item.xpath(f'//div[contains(@class, "yes_heads")][{i+1}]//p/text()[2]')
+                .get()
+                .replace("HOURS: ", "")
+            )
+            oh = OpeningHours()
+            for day in DAYS:
+                if hour == "24/7":
+                    hour = "12:00am - 12:00am"
+                h_h = re.split(" - | -", hour)
+                oh.add_range(
+                    day=day,
+                    open_time=datetime.strftime(
+                        datetime.strptime(
+                            h_h[0].replace("am", " am").replace("pm", " pm"),
+                            "%I:%M %p",
+                        ),
+                        "%H:%M",
+                    ),
+                    close_time=datetime.strftime(
+                        datetime.strptime(
+                            h_h[1].replace("am", " am").replace("pm", " pm"),
+                            "%I:%M %p",
+                        ),
+                        "%H:%M",
+                    ),
```

Comment:
Is there a reason to do `strftime` and `strptime`? I've not tested it, but it seems like the following should work.

```suggestion
                    open_time = h_h[0],
                    close_time = h_h[1],
                    time_format = "%I:%M%p"
```

Reply:
Thanks for your suggestion